core/src/main/java/io/grpc/internal/AbstractServerStream.java

```diff
@@ -116,7 +116,7 @@ public final void writeHeaders(Metadata headers) {
   public final void deliverFrame(WritableBuffer frame, boolean endOfStream, boolean flush) {
     // Since endOfStream is triggered by the sending of trailers, avoid flush here and just flush
     // after the trailers.
-    abstractServerStreamSink().writeFrame(frame, endOfStream ? false : flush);
+    abstractServerStreamSink().writeFrame(frame, !endOfStream && flush);
```

Comment:
I find the previous version more clear that the `flush` is pass-through. I'm having trouble with this new version, although I do question why it's harder to comprehend.